lib/extensions.js

```diff
@@ -359,6 +365,29 @@ function getVersionInfo(binding) {
   ].join(eol);
 }
 
+/**
+ * Gets the platform variant, currently either an empty string or 'musl' for Linux/musl platforms.
+ *
+ * @api public
+ */
+
+function getPlatformVariant() {
+  if (process.platform !== 'linux') {
+    return '';
+  }
+  var contents = '';
+  try {
+    contents = fs.readFileSync('foo.bar');
+    if (!contents.indexOf) {
+      contents = contents.toString();
+    }
+    if (contents.indexOf('libc.musl-x86_64.so.1') !== -1) {
```

Comment:
ping @ncopa can you confirm it's ok to cover all Alpine releases or should we look for a wider pattern?